lib/phlex/elements.rb

```diff
@@ -59,6 +59,11 @@ def #{method_name}(**attributes, &block)
 						buffer << "<#{tag}" << (Phlex::ATTRIBUTE_CACHE[attributes] ||= __attributes__(attributes)) << ">"
 						yield_content(&block)
 						buffer << "</#{tag}>"
+					elsif content
+						buffer << "<#{tag}" << (Phlex::ATTRIBUTE_CACHE[attributes] ||= __attributes__(attributes)) << ">"
+						inline_block = content.is_a?(Proc) ? content : proc { content }
+						yield_content(&inline_block)
```

Comment:
I think we should probably do something more like
```ruby
case content
when Block
  yield_content(&content)
when String
  plain(content)
else
  raise ArgumentError
end
```

Reply:
Good catch. 
Updated. 
There is no `Block` class in ruby. all `proc` and `lambda` is a `Proc`

I moved processing method to `SGML` class. I hope it's ok. 
Tests also improved

Reply:
Pls let me know if I need to update documentation somewhere. Thanks